lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_dns
+short_description: Manage DNS configuration of an ESXI host system
+description:
+- This module can be used to configure DNS for ESXi host systems in given vCenter infrastructure.
+- It's only possible to configured DNS for the default TCP/IP stack.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  host_name:
+    description:
+    - Host name
+    type: str
+    required: True
+  domain:
+    description:
+    - Domain
+    type: str
+    required: True
+    aliases: ['domain_name']
+  dns_servers:
+    description:
+    - DNS servers
+    type: list
+    required: True
+  search_domains:
+    description:
+    - Search domains
+    type: list
+    required: False
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This is required parameter if C(cluster_name) is not specified.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This is required parameter if C(esxi_hostname) is not specified.
+    type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Configure DNS for an host system
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    host_name: '{{ inventory_hostname_short }}'
+    domain: 'example.com'
+    dns_servers:
+      - 192.168.1.10
+      - 192.168.1.11
+    search_domains:
+      - subdomain.example.com
+      - example.com
+  delegate_to: localhost
+
+- name: Configure DNS for all host systems from cluster
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+    domain: 'example.com'
```

Comment:
No need to use quotes here. In the Docs WG we are going to discuss a set of rules for YAML examples in documentation, and I hope we can settle for a minimal YAML syntax, where we only indent as much as is absolutely needed, and only quote where it is required. This way people learn the rules from looking at the examples.

Reply:
Ok. I use quotes for all of my string values. Sometimes it's required (e.g. "{{ variable }}") and sometimes not. I can avoid those annoying errors when I use quotes all the time and don't need to care when I have to use them and when I don't need. So, is there a problem using quotes?

Reply:
ok